app/src/main/java/org/oppia/android/app/spotlight/SpotlightFragment.kt

```diff
@@ -0,0 +1,430 @@
+          Circle((spotlightTarget.anchorHeight / 2).toFloat())
+        } else {
+          Circle((spotlightTarget.anchorWidth / 2).toFloat())
+        }
+      }
+    }
+  }
+
+  private fun getArrowWidth(): Float {
+    return this.resources.getDimension(R.dimen.arrow_width)
+  }
+
+  private fun getArrowHeight(): Float {
+    return this.resources.getDimension(R.dimen.arrow_height)
+  }
+
+  private fun getScreenCentreY(): Int {
+    Log.d("overlay screenCentre Y", (screenHeight / 2).toString())
+    return screenHeight / 2
+  }
+
+  private fun getScreenCentreX(): Int {
+    Log.d("overlay screenCentre X", (screenWidth / 2).toString())
+    return screenWidth / 2
+  }
+
+  private fun calculateAnchorPosition(spotlightTarget: SpotlightTarget): AnchorPosition {
+    Log.d("overlay", "checking anchor position. RTL = $isRtl")
+    return if (spotlightTarget.anchorCentreX > getScreenCentreX()) {
+      if (spotlightTarget.anchorCentreY > getScreenCentreY()) {
+        AnchorPosition.BottomRight
+      } else {
+        AnchorPosition.TopRight
+      }
+    } else if (spotlightTarget.anchorCentreY > getScreenCentreY()) {
+      AnchorPosition.BottomLeft
+    } else {
+      AnchorPosition.TopLeft
+    }
+  }
+
+  private fun requestOverlayResource(spotlightTarget: SpotlightTarget): View {
+    val anchorPosition = calculateAnchorPosition(spotlightTarget)
+    Log.d("overlay", anchorPosition.toString())
+
+    return when (anchorPosition) {
+      AnchorPosition.TopLeft -> {
+        if (isRtl) {
+          configureTopRightOverlay(spotlightTarget)
+        } else {
+          configureTopLeftOverlay(spotlightTarget)
+        }
+      }
+      AnchorPosition.TopRight -> {
+        if (isRtl) {
+          configureTopLeftOverlay(spotlightTarget)
+        } else {
+          configureTopRightOverlay(spotlightTarget)
+        }
+      }
+      AnchorPosition.BottomRight -> {
+        if (isRtl) {
+          configureBottomLeftOverlay(spotlightTarget)
+        } else {
+          configureBottomRightOverlay(spotlightTarget)
+        }
+      }
+      AnchorPosition.BottomLeft -> {
+        if (isRtl) {
+          configureBottomRightOverlay(spotlightTarget)
+        } else {
+          configureBottomLeftOverlay(spotlightTarget)
+        }
+      }
+    }
+  }
+
+  private sealed class AnchorPosition {
+    /** The position corresponding to the anchor when it is on the top left of the screen. */
+    object TopLeft : AnchorPosition()
+    /** The position corresponding to the anchor when it is on the top right of the screen. */
+    object TopRight : AnchorPosition()
+    /** The position corresponding to the anchor when it is on the bottom left of the screen. */
+    object BottomLeft : AnchorPosition()
+    /** The position corresponding to the anchor when it is on the bottom right of the screen. */
+    object BottomRight : AnchorPosition()
+  }
+
+  override fun clickOnNextTip() {
```

Comment:
Similarly here, the class organization is a bit confusing. I suggest ordering as so:
- Constants
- Public vals/vars
- Private vals/vars
- Override methods
- Public non-override methods
- Private methods
- Public classes
- Private classes
- Companion object

Reply:
done